graylog2-web-interface/src/stores/field-analyzers/FieldQuickValuesStore.js

```diff
@@ -0,0 +1,78 @@
+import Reflux from 'reflux';
+
+import URLUtils from 'util/URLUtils';
+import ApiRoutes from 'routing/ApiRoutes';
+import fetch from 'logic/rest/FetchProvider';
+import UserNotification from 'util/UserNotification';
+
+import CombinedProvider from 'injection/CombinedProvider';
+
+const { SearchStore } = CombinedProvider.get('Search');
+const { FieldQuickValuesActions } = CombinedProvider.get('FieldQuickValues');
+
+const FieldQuickValuesStore = Reflux.createStore({
+  listenables: [FieldQuickValuesActions],
+
+  get(field, options = {}) {
+    const { order, tableSize, stackedFields } = options;
+
+    this.trigger({ loading: true });
+    const originalSearchURLParams = SearchStore.getOriginalSearchURLParams();
+    const streamId = SearchStore.searchInStream ? SearchStore.searchInStream.id : null;
+
+    const rangeType = originalSearchURLParams.get('rangetype');
+    const timerange = {};
+    switch (rangeType) {
+      case 'relative':
+        timerange.range = originalSearchURLParams.get('relative');
+        break;
+      case 'absolute':
+        timerange.from = originalSearchURLParams.get('from');
+        timerange.to = originalSearchURLParams.get('to');
+        break;
+      case 'keyword':
+        timerange.keyword = originalSearchURLParams.get('keyword');
+        break;
+      default:
+      // Do nothing
+    }
+
+    const url = ApiRoutes.UniversalSearchApiController.fieldTerms(
+      rangeType,
+      originalSearchURLParams.get('q') || '*',
+      field,
+      order,
+      tableSize,
+      stackedFields,
+      timerange,
+      streamId,
+    ).url;
+
+    const promise = fetch('GET', URLUtils.qualifyUrl(url));
+    promise.then(
+      (response) => {
+        this.trigger({ data: response, loading: false });
+      },
+      this._errorHandler('Loading quick values failed with message', 'Could not load quick values'),
+    );
+
+    FieldQuickValuesActions.get.promise(promise);
+  },
+
+  _errorHandler(message, title, cb) {
+    return (error) => {
+      let errorMessage;
+      try {
+        errorMessage = error.additional.body.message;
```

Comment:
Are we always going to set a user-friendly body message on responses to this request? I know the way we handle errors is far from good, but this pattern we repeat in many places (also duplicating code all over the place) is also not that great. Error messages returned from the server are often too technical and obscure, and other times too vague for being shown to a non-technical operator.

To put an example, and I know we discussed this before. Selecting a stacked field in ES 2.x returns: `Loading quick values failed with message: Unable to perform terms query script_lang not supported [painless]` which is not helping me at all to know what went wrong with that, just making me confused.

At least I think we should have this kind of error handling abstracted somewhere and not repeating it in every store.

Reply:
Also now that I look at it, we even accept a callback there for some reason, I guess c&p 😄

Reply:
> Are we always going to set a user-friendly body message on responses to this request?

Nope we don't. I agree that we need some kind of centralized error handling where we can map error codes from the backend to user friendly messages in the UI. (or somthing similar)

Until we have this showing a message like in the code snippet below makes it really hard to debug the problem because our exception handling in REST resources is mostly automatic and does not log anything to the server log. 

```javascript
UserNotification.error(`Loading quick values failed with status: ${error}`,
          'Could not load quick values');
```

So I personally rather have a message that the user might not understand correctly but which helps me to debug the problem. :smiley: 

I completely agree regarding the code duplication. If we want to use this kind of error handling until we have a better one, we should put this into a helper function that can be shared.

Reply:
Those errors may help solving the problem in some cases, but they are overly complicated, sometimes even include some Java exceptions and may even leak some information. Today for instance I was playing today with the email delivery and some details about the email server and so on may be leak from the API really easily without knowing it. As I said before, in my opinion that is no error message for a non-technical operator, which may operate Graylog.

Anyway, if we really want to change that behaviour or abstract it, we already have a helper for it:
https://github.com/Graylog2/graylog2-server/blob/master/graylog2-web-interface/src/logic/rest/FetchProvider.js#L14

We could add a method there without changing the default behaviour to make easier to access the body error message if it's available.

Reply:
Okay, I removed the custom error handler. This needs a centralized solution. :smiley:

Reply:
Hehe, ok. I'll add a task to my to do to at least discuss it or make it easier to access the response message. 👍